src/software/thunderscope/gl/gl_world_layer.py

```diff
@@ -0,0 +1,482 @@
+                [[0, -(field.field_y_length / 2)], [0, (field.field_y_length / 2)]]
+            ),
+        )
+
+        # Goal-to-goal line
+        field_marking_line_graphics[1].setData(
+            pos=np.array(
+                [[-(field.field_x_length / 2), 0], [(field.field_x_length / 2), 0]]
+            ),
+        )
+
+        # Center circle
+        field_center_circle_graphic.set_radius(field.center_circle_radius)
+
+    def __update_goal_graphics(self, field: Field):
+        """Update the GLGraphicsItems that display the goals
+        
+        :param field: The field proto
+
+        """
+        goal_graphics = self.graphics_list.get_graphics("goals", 2)
+
+        # Friendly goal
+        goal_graphics[0].set_dimensions(field.goal_x_length, field.goal_y_length)
+        goal_graphics[0].set_position(-field.field_x_length / 2, 0)
+        goal_graphics[0].set_orientation(0)
+
+        # Enemy goal
+        goal_graphics[1].set_dimensions(field.goal_x_length, field.goal_y_length)
+        goal_graphics[1].set_position(field.field_x_length / 2, 0)
+        goal_graphics[1].set_orientation(180)
+
+    def __update_ball_graphics(self, ball_state: BallState):
+        """Update the GLGraphicsItems that display the ball
+        
+        :param ball_state: The ball state proto
+
+        """
+        # Update the ball graphic
+        ball_graphic = self.graphics_list.get_graphics("ball", 1)[0]
+        ball_graphic.set_position(
+            ball_state.global_position.x_meters,
+            ball_state.global_position.y_meters,
+            ball_state.distance_from_ground,
+        )
+
+    def __update_robot_graphics(self, team: Team, color):
+        """Update the GLGraphicsItems that display the robots
+        
+        :param team: The team proto
+        :param color: The color of the robots
+
+        """
+        for robot_graphic, robot_id_graphic, robot in zip(
+            self.graphics_list.get_graphics("robots", len(team.team_robots)),
+            self.graphics_list.get_graphics("robot_ids", len(team.team_robots)),
+            team.team_robots,
+        ):
+            robot_graphic.set_position(
+                robot.current_state.global_position.x_meters,
+                robot.current_state.global_position.y_meters,
+            )
+            robot_graphic.set_orientation(
+                math.degrees(robot.current_state.global_orientation.radians)
+            )
+            robot_graphic.setColor(color)
+
+            if self.display_robot_id:
+                robot_id_graphic.show()
+                robot_id_graphic.setData(
+                    text=str(robot.id),
+                    pos=[
+                        robot.current_state.global_position.x_meters
+                        + (ROBOT_MAX_RADIUS_METERS / 2),
+                        robot.current_state.global_position.y_meters,
+                        ROBOT_MAX_HEIGHT_METERS + 0.1,
+                    ],
+                )
```

Comment:
Trying to position text right in the middle of the bot is difficult since text can only be left-aligned and its font size doesn't scale as we zoom in (it's a billboard)

When zoomed in the IDs appear centered:
![image](https://github.com/UBC-Thunderbots/Software/assets/60044853/dde8d58e-688b-4574-be75-2ed379cc0141)

But the text becomes offset and illegible when zoomed out:
![image](https://github.com/UBC-Thunderbots/Software/assets/60044853/8c3e8dcf-abc1-4145-88d2-627063ad1456)

And the text is off center when looking at it from the side:
![image](https://github.com/UBC-Thunderbots/Software/assets/60044853/3abfeefd-1d71-416b-a6ea-942165d9376d)

We could write a lot of hacky code to calculate the position of the text based on the camera but I don't think that's wise. The latest set of text offsets that I just committed produces this result, which I think looks acceptable from most angles:
![image](https://github.com/UBC-Thunderbots/Software/assets/60044853/9143d817-9f2a-41d1-a634-1e9d4faadba1)